### ISSUE TITLE
no-restricted-imports doesn't handle re-export

### ISSUE THREAD

**Issue by asapach** [eslint/eslint#9678]
* **ESLint Version:** 4.12.1

```js
{
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module"
    },
    "rules": {
        "no-restricted-imports": ["error", "fs"]
    }
}
```

```js
import fs from 'fs';
export * from 'fs';
export {default as fs} from 'fs';
export {readFile} from 'fs';
```

**What did you expect to happen?**
All the statements including re-exports should be reported as errors.

**What actually happened? Please include the actual, raw output from ESLint.**
```bash
  1:1  error  'fs' import is restricted from being used  no-restricted-imports
```
Only import is reported; re-exports are ignored.

**Comment by ilyavolodin** [comment]
The rule is supposed to only check `import` statements, so I'm not sure why would it report an export.

**Comment by j-f1** [comment]
This is because the `export ... from` is [not yet](https://github.com/tc39/proposal-export-ns-from) [part of JavaScript](https://github.com/tc39/proposal-export-default-from), and ESLint doesn’t support features that are still proposals. This would be a good rule suggestion for [`eslint-plugin-babel`](https://github.com/babel/eslint-plugin-babel), though.

**Comment by asapach** [comment]
@ilyavolodin, Because a re-export is essentially an import + export. So `export {readFile} from 'fs'` is equivalent to:
```js
import {readFile} from 'fs';
export {readFile};
```
By ignoring re-exports you are leaving an escape hatch for `no-restricted-imports`, so people can re-export forbidden modules and import them elsewhere.

@j-f1 I'm talking about ES2015 syntax; future proposals are out of scope.

**Comment by j-f1** [comment]
@asapach I’m trying to say that everything but the first line in your initial code sample is **not** part of ES2015, ES2016, or ES2017. It’s currently a [stage 1 proposal](https://github.com/tc39/proposals#stage-1) (`export v from "mod";` statements). Therefore, it’s out-of-scope for ESLint.

However, I would be interested in seeing this checked by `eslint-plugin-babel`.

**Comment by asapach** [comment]
@j-f1, tell that to @nzakas:
* http://2ality.com/2014/09/es6-modules-final.html#re-exporting
* http://exploringjs.com/es6/ch_modules.html#_re-exporting

**Comment by asapach** [comment]
@j-f1 [the proposal](https://github.com/tc39/proposal-export-default-from) is specifically about a more convenient syntax for re-exporting `default` and has nothing to do wit this issue.

**Comment by j-f1** [comment]
🤦‍♂️ Thanks for sharing that with me! I must’ve misread the proposals I linked to. I’ll take a look at this and see if I can fix it.

**Comment by not-an-aardvark** [comment]
@j-f1 There's no rush, but have you gotten a chance to take a look at this?

**Comment by j-f1** [comment]
I have not gotten a chance to look at this yet.
